stellarphot/photometry/photometry.py

```diff
@@ -646,35 +749,37 @@ def multi_image_photometry(directory_with_images,
     # Attributes should survive intact assume all have same camera and observatory.
     all_phot = vstack(phots)
 
-    # Build a set of all the unique star_ids in the sourcelist
-    # that were eliminated on at least one image
-    if len(missing_sources) > 0:
-        if len(missing_sources) > 1:
-            uniques = set(missing_sources)
-        else:
-            uniques = set([missing_sources])
-
-        print("  Removing {len(uniques)} sources not observed "
-              "in every image ... ", end="")
-        # Purge the photometry table of all sources that were eliminated
-        # on at least one image
-        starid_to_remove = sorted([u for u in uniques if u in all_phot['star_id']])
-        # add index to PhotometryData to speed up removal
-        all_phot.add_index('star_id')
-        # Remove the starid for objects not observed in every image
-        if starid_to_remove:
-            bad_rows = all_phot.loc_indices[starid_to_remove]
-            try:
-                bad_rows = list(bad_rows)
-            except TypeError:
-                bad_rows = [bad_rows]
+    # If requested, eliminate source not detected on every image by building
+    # a set of all the unique star_ids that were missing on at least one image.
+    if (reject_unmatched):
```

Comment:
Could we rewrite this block to start with this so we can reduce the indent level?

```suggestion
    if reject_unmatched and (len(missing_sources) > 0):
```

Reply:
Good catch, fixed.